ssh/tailssh/incubator.go

```diff
@@ -370,26 +373,26 @@ func dropPrivileges(logf logger.Logf, wantUid, wantGid int, supplementaryGroups
 	// cannot reset the it back to our original values, and that the
 	// current egid/euid are the expected values after we change
 	// everything; if not, we exit the process.
-	if assertDropPrivileges {
+	if assertPrivilegesWereDroppedByAttemptingToUnDrop {
 		if egid != wantGid {
 			if err := syscall.Setegid(egid); err == nil {
-				fatalf("unexpectedly able to set egid back to %d", egid)
+				fatalf("able to set egid back to %d", egid)
 			}
 		}
 		if euid != wantUid {
 			if err := syscall.Seteuid(euid); err == nil {
-				fatalf("unexpectedly able to set euid back to %d", euid)
+				fatalf("able to set euid back to %d", euid)
 			}
 		}
-
+	}
+	// TODO(andrew): remove this envknob after sufficient testing, e.g. the 1.40 release
+	if envknob.BoolDefaultTrue("TS_DEBUG_INCUBATOR_ASSERT_PRIVILEGES") {
```

Comment:
can you make sure that this is never forwarded from the SSH session?
Actually, maybe lets not have an envknob here at all.

Reply:
I could just unset the variable unconditionally after this `if`?

Reply:
the incubator doesn't get any of the `os.Environ`. So you'll have to build a custom binary anyways, so a const should work?

Reply:
Hm, yeah, a `const` is probably fine then

Reply:
Pushed; PTAL?